core/src/instance.rs

```diff
@@ -48,6 +51,11 @@ impl Instance {
         let (rx_action, rx_observer) = self.initialize_channels();
         let context = self.initialize_context(context);
         self.start_action_loop(context.clone(), rx_action, rx_observer);
+        let mut scheduler = Scheduler::new();
+        scheduler
+            .every(10.seconds())
+            .run(scheduled_jobs::create_callback(context.clone()));
+        self.scheduler_handle = Some(Arc::new(scheduler.watch_thread(Duration::from_millis(5))));
```

Comment:
5ms seems pretty short to be checking if 10 minutes have passed yet, am I reading this right? The interval isn't so important, this could easily be 1 sec to give us retries every 9-11 seconds, right?

Reply:
it's checking if 10 seconds have passed not 10 minutes, but yeah...